cmake/modules/clang-format-check.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
```

Comment:
FWIW there is a project I am involved in where the style is enforced as follows:

```bash
clang-format -i -fallback-style=none -style=file $all_source_files || exit 1
modified="$(git status --porcelain --untracked-files=no)"
if [ -n "$modified" ]; then echo "Run enforce_style.sh to reformat the code."; exit 1; fi
```

It doesn't fit the description "no source code is modified", but as we're doing that in the CI environment we don't really care. Just sharing here.

Reply:
`-fallback-style=none` is something I should be doing but otherwise I like my solution which does not modify the source. 

No sooner that I had written this then I was made aware of a better OSS version (https://github.com/Sarcasm/run-clang-format). But mine works and I wrote it (god-dammit!).